src/main/java/org/magicdgs/thaplv/tools/GetHaplotypes.java

```diff
@@ -0,0 +1,123 @@
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.HaploidWalker;
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilterLibrary;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+
+import java.io.File;
+
+/**
+ * Simple tool to convert haplotypes (either Inbred Lines or Back-crosses) to the haploid format
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+@CommandLineProgramProperties(
+        summary =
+                "Check diploid calls obtained with other callers and get the haploid/haplotypes using the provided model."
+                        + "It allows to don't use any model afterwards (DONT_CHECK) to speed up the analysis without conversion.",
+        oneLineSummary = "Get the haplotypes from a backcross or inbred line experiment.",
+        programGroup = AlphaProgramGroup.class
+)
+public class GetHaplotypes extends HaploidWalker {
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "File to which variants should be written", common = false, optional = false)
+    public File outFile;
+
+    @Argument(fullName = ThaplvArgumentDefinitions.ONLY_POLYMORPHIC_LONG, shortName = ThaplvArgumentDefinitions.ONLY_POLYMORPHIC_SHORT, doc = "Only polymorphic sites will be output.", optional = true)
+    public boolean onlyPolymorphic = false;
+
+
+    @Override
+    protected boolean requiresOutputPloidy() {
+        return true;
+    }
+
+    @Override
+    protected boolean allowsDontCheck() {
+        return false;
+    }
+
+    private VariantContextWriter vcfWriter = null;
+
+    /**
+     * Make the variant filter to filter invariant sites
+     */
+    protected VariantFilter makeVariantFilter() {
+        if (onlyPolymorphic) {
+            return HaplotypeFilterLibrary.POLYMORPHIC_FILTER;
+        }
+        return VariantFilterLibrary.ALLOW_ALL_VARIANTS;
+    }
+
+    /**
+     * Set up the VCF writer
+     */
+    @Override
+    public void onTraversalStart() {
+        // Initialize VCF header lines
+        final VCFHeader header = getHeaderForVariants();
+        header.addMetaDataLine(new VCFHeaderLine("source", this.getClass().getSimpleName()));
+        vcfWriter = GATKVariantContextUtils
+                .createVCFWriter(outFile, getBestAvailableSequenceDictionary(), false);
+        vcfWriter.writeHeader(header);
+    }
+
+    @Override
+    public void apply(VariantContext variant, ReadsContext readsContext,
+            ReferenceContext referenceContext, FeatureContext featureContext) {
+        // TODO: convert some of the annotations for the VCF for something
```

Comment:
Probably open a ticket with low priority for this? And it will be better if an option is added for this behavior.